pennylane/templates/subroutines/hilbert_schmidt.py

```diff
@@ -105,6 +105,16 @@ def _flatten(self):
         )
         return self.data, metadata
 
+    @classmethod
+    def _primitive_bind_call(cls, *args, **kwargs):
+        # This is more complicated than it would need to be because of
+        # the three keyword-only arguments of __init__.
+        for kwarg_name in ("u_tape", "v_wires", "v_function"):
+            if kwarg_name not in kwargs:
+                kwargs[kwarg_name] = args[-1]
+                args = args[:-1]
+        return cls._primitive.bind(*args, **kwargs)
```

Comment:
```suggestion
    def _primitive_bind_call(cls, *params, v_function, v_wires, u_tape, id=None):
        return cls._primitive.bind(*params, v_function=v_function, v_wires=v_wires, u_tape=u_tape, id=id)
```

This might be a good use-case for writing out the call signature. Not sure how this'll work with `v_wires` being metadata here, but I don't think we need to be too concerned about the beauty of this operation.

Reply:
Ah of course, that's so much nicer! I guess `u_tape` being metadata could also lead to problems, like `v_wires` :thinking: